pkgs/development/python-modules/adafruit-nrfutil/default.nix

```diff
@@ -0,0 +1,53 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pythonOlder
+, pyserial
+, click
+, ecdsa
+, behave
+, nose
+}:
+
+buildPythonPackage rec {
+  pname = "adafruit-nrfutil";
+  version = "0.5.3.post17";
+  format = "setuptools";
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "adafruit";
+    repo = "Adafruit_nRF52_nrfutil";
+    rev = "1361059009ff6a24d63b37eb3a4b28127837ead2";
```

Comment:
```suggestion
    rev = version;
```

Reply:
The tag `0.5.3.post17` does not point to a version containing the fixes for the required tests (https://github.com/adafruit/Adafruit_nRF52_nrfutil/pull/38/commits/e5fbcc8ee5958041db38c04139ba686bf7d1b845). The maintainer decided against releasing a new version, see https://github.com/adafruit/Adafruit_nRF52_nrfutil/pull/38#issuecomment-1169876952 .

Reply:
We can't just fetch a newer commit and leave the version as is. I would suggest to download the tagged version and fetch the patch we need.

Reply:
Fair point. I adjusted the code to download the PR.